java/vector/src/main/java/org/apache/arrow/vector/compare/RangeEqualsVisitor.java

```diff
@@ -49,38 +56,66 @@ public RangeEqualsVisitor(ValueVector right, int leftStart, int rightStart, int
     this.rightStart = rightStart;
     this.right = right;
     this.length = length;
+    Preconditions.checkArgument(length >= 0, "length must be non negative");
+  }
+
+  /**
+   * Reset start indices and length for reuse purpose.
+   */
+  public void reset(int leftStart, int rightStart, int length) {
```

Comment:
I think it's better to just create a new `RangeEqualsVisitor` than add a `reset` method

Reply:
removed.